i2c-hid.c

```diff
@@ -807,7 +807,7 @@ static int i2c_hid_init_irq(struct i2c_client *client)
 	dev_dbg(&client->dev, "Requesting IRQ: %d\n", client->irq);
 
 	if (!irq_get_trigger_type(client->irq))
```

Comment:
In this particular case, you want to remove the `if`.
The irq should be already set by i2c-core/ACPI, so in this case, you code doesn't change a bit :)

Reply:
Woops, a great newbie mistake !

But that does not change the device problem...